SA3/qa_web/views.py

```diff
@@ -147,3 +155,178 @@ def vote(request):
 # Home Page
 def homepage(request):
     return render(request, "qa_web/home.html")
+
+
+class QuestionDisplayView(ListView):
+    model = Questions
+    paginate_by = 10  # 10 questions a page
+    context_object_name = 'questions'
+    template_name = 'qa_web/question_display_page.html'
+    ordering = '-creation_date'
+
+    def get_context_data(self, *args, **kwargs):
+        context = super(
+            QuestionDisplayView, self).get_context_data(*args, **kwargs)
+
+        context['total_question_num'] = Questions.objects.count()
+        context['total_answer_num'] = Answers.objects.count()
+
+        #  try to page the question page
+        context['is_paginated'] = True  # whether the paginator is needed. It's a must.
+        all_questions_qs = Questions.objects.order_by('-creation_date')\
+            .select_related('owner')\
+            .annotate(num_answers=Count('answers', distinct=True),
+                      num_question_comments=Count('comments',
+                      distinct=True))
```

Comment:
Cool implementation of the count of answers to question and comments to question through `Count`.